### PR TITLE
chore: update `AccountProfile` and related components

### PR DESCRIPTION
**Pull request by raastinn**
## Compatiblity

Any changes that require a newer version of the API are considered breaking
changes.

Common breaking changes:
 - Making a request to a new API endpoint that will return `403` or `404` in older versions of the API.
 - No longer sending request fields required by previous versions of the API 
 - Requiring new fields from an API endpoint that will not be present in
   responses from older versions of the API
 - Breaking backwards compatibility with old response data shapes

Choose one:
 - [x] The changes do not break compatibility
 - [ ] The changes potentially break compatibility

## Pre-Review Checklist
 - [x] All changes are tested
 - [x] All touched code documentation is updated
 - [x] All tests pass in your local environment
 - [x] Deepsource issues have been reviewed and addressed
 - [x] Debugging logging and commented out code has been removed

 
## Screenshots:
_Only required for visual changes_.

### CODE REVIEW

File: src/js/account/api.ts [hunk 1]
```diff
@@ -26,8 +27,10 @@ export function get(): Promise<Response> {
  * @returns A promise resolving to a response containing the updated
  * user's account data
  */
-export function update({ update }): Promise<Response> {
-    return Request.patch("/account").send(update);
+export function updateAccount(update: string): Promise<User> {
```
Comment (ReeceHoffmann) [inline src/js/account/api.ts]: Is that typing for update correct? I would assume that any update needs to be a key:value pair

File: src/js/account/components/AccountProfile.tsx [hunk 2]
```diff
@@ -39,47 +36,37 @@ const AdministratorTag = styled(Label)`
...
  */
-function AccountProfile({ administratorRole, groups, handle }: AccountProfileProps) {
+export default function AccountProfile() {
+    const { data, isLoading } = useFetchAccount();
+
+    if (isLoading) {
+        return <LoadingPlaceholder />;
+    }
+
+    const { administrator_role, groups, handle } = data;
```
Comment (ReeceHoffmann) [inline src/js/account/components/AccountProfile.tsx]: total nit:

Stylistic choice, pretty arguable, but i would just extract all the variables you need here. Saves having to do the `data.whatever` syntax below.

File: src/js/base/InputPassword.tsx [hunk 3]
```diff
@@ -1,15 +1,20 @@
...
+};
+
+export const InputPassword = React.forwardRef<HTMLInputElement, InputPasswordProps>((props, ref) => {
     const [show, setShow] = useState(false);
 
     return (
-        <InputContainer align="right">
-            <Input name="password" aria-label="password" type={show ? "text" : "password"} {...props} />
+        <InputContainer>
+            <Input as="input" {...props} ref={ref} type={show ? "" : "password"} />
```
Comment (ReeceHoffmann) [inline src/js/base/InputPassword.tsx:16]: Why the `as`? Shouldn't the underlying type of `Input` be `input`?

**Review by ReeceHoffmann — CHANGES_REQUESTED** [review]

File: src/js/base/InputPassword.tsx (hunk 3 above)
Comment (raastinn) [reply src/js/base/InputPassword.tsx:16]: Not sure why this was happening, but without the `as="input"`, the input wouldn't register the text from react-hook-form.

Comment (ReeceHoffmann) [reply src/js/base/InputPassword.tsx:16]: Hmm interesting, is `Input` maybe not forwarding the ref to the html `input`? I think this is worth fixing, having to overwrite with `as` seems kinda brutal

**Review by ReeceHoffmann — APPROVED** [review]

**Pull request merged**